### ISSUE TITLE
Training P-Net

### ISSUE THREAD

**Issue by SURABHI-GUPTA** [look4pritam/tfmtcnn#16]
With 19 epochs, the loss doesn't seem to converge. What can be the issue ?

**Comment by look4pritam** [comment]
loss decreases in first 5-10 epochs, and then keep on optimizing or
toggling.

**Comment by SURABHI-GUPTA** [comment]
have you evaluated the performance on wider-val easy/medium/hard ?

**Comment by look4pritam** [comment]
I have evaluated performance on wider val dataset, it is similar to
published paper. I am not able ti recall exact numbers. In real environment
also performance is good.
For low resolution images, i am using mtcnn.
But i am using Yolo-v3 for high resolution images, like full hd.
Yolo-v3 provides much better performance with constant fps irrespective of
resolution.

**Comment by SURABHI-GUPTA** [comment]
have you used the matlab code ? Because my precision recall curve is giving very low numbers

**Comment by look4pritam** [comment]
no i have not used matlab code. I have used mtcnn code and yolo-v3 code.

**Comment by look4pritam** [comment]
I have used
https://github.com/sthanhng/yoloface
weights for yolo-v3.

**Comment by look4pritam** [comment]
use mtcnn accurate model and evaluation results are
https://github.com/look4pritam/tfmtcnn/blob/master/tfmtcnn/models/mtcnn/accurate/output/evaluate_model.out

**Comment by SURABHI-GUPTA** [comment]
yes.. I tried.. but they are not giving similar results

**Comment by SURABHI-GUPTA** [comment]
I tried every model of yours but it is not giving me similar result on wider val

**Comment by look4pritam** [comment]
If you are using OpenCV for image reading then convert opencv bgr image to
rgb format. the models are trained for rgb images. It changes the
performance.

**Comment by SURABHI-GUPTA** [comment]
I have used your webcam-demo.py to detect bbx. So, the code already implements this. I am taking minface size= 20 and threshold as 0.8.

**Comment by look4pritam** [comment]
It is evaluated on FDDB dataset. It is not evaluated on WIDER dataset splits.
minimum face size is 100, default threshold is 0.8, but you can try 0.7 also.

**Comment by SURABHI-GUPTA** [comment]
okay.. thanks

**Comment by look4pritam** [comment]
If you want speed use mtcnn.
If yoy want accuracy and your input is high resolution use yolo-v3.
If you want best accuracy and speed is not a problem, then use retinaface.

**Comment by SURABHI-GUPTA** [comment]
thanks for the info

**Comment by SURABHI-GUPTA** [comment]
one more question, have you implemented online hard mining ?

**Comment by look4pritam** [comment]
yes

**Comment by look4pritam** [comment]
See
https://github.com/look4pritam/tfmtcnn/blob/master/tfmtcnn/datasets/HardFaceDataset.py

Here model is small, so gets overfitted, that needs to be stopped.
I have done this using this evaluation on fddb manually.
